src/javascript/app_2/pages/trading/components/data_table.jsx

```diff
@@ -0,0 +1,347 @@
+        const { displayData } = this.state;
+
+        return (
+            <div className='table-container'>
+                <table className='table'>
+                    <thead className='table-thead'>
+                        <tr className='table-row'>
+                            {this.renderHeaders()}
+                        </tr>
+                    </thead>
+
+                    <tbody className='table-tbody'>
+                        {this.renderBodyRows()}
+                    </tbody>
+                </table>
+
+                {pagination && this.renderPagination()}
+            </div>
+        );
+    }
+}
+
+DataTable.defaultProps = {
+    pagination: true,
+    pagesCloseToEnd: 5
+};
+
+export class StatementDataTable extends React.PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.getNextBatch = this.getNextBatch.bind(this);
+
+        const columns = [
+            {
+                title: localize('Date'),
+                dataIndex: 'date',
+            },
+            {
+                title: localize('Ref.'),
+                dataIndex: 'ref',
+            },
+            {
+                title: localize('Potential Payout'),
+                dataIndex: 'payout',
+            },
+            {
+                title: localize('Action'),
+                dataIndex: 'action',
+            },
+            {
+                title: localize('Description'),
+                dataIndex: 'desc',
+            },
+            {
+                title: localize('Credit/Debit'),
+                dataIndex: 'amount',
+                renderCell: (data, dataIndex, transaction) => {
+                    const parseStrNum = (str) => {
+                        return parseFloat(str.replace(',', '.'));
+                    }
+                    return <td className={`${dataIndex} ${(parseStrNum(data) >= 0) ? 'profit' : 'loss'}`} key={dataIndex}>{data}</td>
+                }
+            },
+            {
+                title: localize('Balance'),
+                dataIndex: 'balance',
+            }
+        ];
+
+        this.state = {
+            dataSource: [],
+            columns: columns,
+            loadedAllTransactions: false
+        };
+    }
+
+    getStatementData(statement, currency, jp_client) {
+        const date_obj   = new Date(statement.transaction_time * 1000);
+        const moment_obj = moment.utc(date_obj);
+        const date_str   = moment_obj.format('YYYY-MM-DD');
+        const time_str   = `${moment_obj.format('HH:mm:ss')} GMT`;
+        const payout     = parseFloat(statement.payout);
+        const amount     = parseFloat(statement.amount);
+        const balance    = parseFloat(statement.balance_after);
+        const is_ico_bid = /binaryico/i.test(statement.shortcode);
+
+        let action = localize(toTitleCase(statement.action_type));
+        if (is_ico_bid) {
+            action = /buy/i.test(statement.action_type) ? localize('Bid') : localize('Closed Bid');
```

Comment:
i think we'll be removing ICO soon, it's probably safe to remove ICO related code